Kickstarter-iOS/Views/Controllers/ProjectPageViewController.swift

```diff
@@ -0,0 +1,415 @@
+import KsApi
+import Library
+import Prelude
+import UIKit
+
+public final class ProjectPageViewController: UIViewController, MessageBannerViewControllerPresenting {
+  // MARK: Properties
+
+  private enum NavigationButtonSizes: CGFloat {
+    case spacing = 15.0
+  }
+
+  private let viewModel: ProjectPageViewModelType = ProjectPageViewModel()
+  private let shareViewModel: ShareViewModelType = ShareViewModel()
+  private let watchProjectViewModel: WatchProjectViewModelType = WatchProjectViewModel()
+  /**
+   FIXME: This `contentController` can be renamed `contentViewController` and has to be embedded in a `PagingViewController` in https://kickstarter.atlassian.net/browse/NTV-195
+   Maybe check `BackerDashboardViewController`'s pageViewController for in-app examples on how to do this.
+   */
+  private var contentController: ProjectPamphletContentViewController?
+
+  // MARK: User session state properties
+
+  private var sessionEndedObserver: Any?
+  private var sessionStartedObserver: Any?
+
+  public var messageBannerViewController: MessageBannerViewController?
+
+  private lazy var navigationShareButton: UIBarButtonItem = {
+    let contentView = UIButton()
+      |> shareButtonStyle
+      |> UIButton.lens.imageEdgeInsets .~ UIEdgeInsets(left: -NavigationButtonSizes.spacing.rawValue)
+      |> UIButton.lens.accessibilityLabel %~ { _ in Strings.dashboard_accessibility_label_share_project() }
+
+    contentView.addTarget(
```

Comment:
Most of the time I see the styling done in a `bindStyles` method and the `addTarget` being done in `viewDidLoad`. I don't know if we also do it like this but I bet it's for simpler initialization and easier readability up top.